[bug]  [Meson toolchain failing tests] on GNU Guix

### Environment Details (include every applicable attribute)
  * Operating System+version: GNU Guix, main branch
  * Compiler+version: GCC 7.5.0
  * Conan version: 1.40.2
  * Python version: 3.8.2

### Steps to reproduce (Include if Applicable)

Build a newly defined conan package for Guix, which is built in a container with the the following direct dependencies:

```
autoconf@2.69 automake@1.16.2 cmake@3.21.1 coreutils@8.32 git-minimal@2.33.0 meson@0.55.1 ninja@1.10.0 pkg-config@0.29.2 python-bottle@0.12.19 python-colorama@0.4.4 python-dateutil@2.8.1
+ python-distro@1.4.0 python-fasteners@0.15 python-future@0.17.1 python-jinja2@2.11.2 python-mock@3.0.5 python-node-semver@0.8.1 python-parameterized@0.7.4 python-patch-ng@1.17.4 python-pluginbase@1.0.1
+ python-pygments@2.7.3 python-pyjwt@1.7.1 python-pytest-xdist@1.34.0 python-pytest@5.3.5 python-pyyaml@5.4.1 python-requests@2.25.0 python-six@1.14.0 python-tqdm@4.43.0 python-urllib3@1.26.7
+ python-webtest@2.0.35 which@2.21
```

### Logs (Executed commands with output) (Include/Attach if Applicable)

Some failures are ignored because they occur due to networking or other assumptions about the system that doesn't hold in Guix, but I couldn't find an explanation for these two:

```
_________________ MesonPreprocessorDefinitionsTest.test_build __________________
[gw0] linux -- Python 3.8.2 /gnu/store/f8s95qc6dfhl0r45m70hczw5zip0xjxq-python-wrapper-3.8.2/bin/python

self = <conans.test.functional.toolchains.meson.test_preprocessor_definitions.MesonPreprocessorDefinitionsTest testMethod=test_build>

    def test_build(self):
        hello_h = gen_function_h(name="hello")
        hello_cpp = gen_function_cpp(name="hello",
                                     preprocessor=["TEST_DEFINITION1", "TEST_DEFINITION2"])
        app = gen_function_cpp(name="main", includes=["hello"], calls=["hello"])
    
        self.t.save({"conanfile.py": self._conanfile_py,
                     "meson.build": self._meson_build,
                     "hello.h": hello_h,
                     "hello.cpp": hello_cpp,
                     "main.cpp": app})
    
        self.t.run("install . %s" % self._settings_str)
    
        content = self.t.load("conan_meson_native.ini")
    
        self.assertIn("[built-in options]", content)
        self.assertIn("buildtype = 'release'", content)
    
>       self.t.run("build .")

conans/test/functional/toolchains/meson/test_preprocessor_definitions.py:60: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
conans/test/utils/tools.py:568: in run
    error = self.run_cli(command_line, assert_error=assert_error)
conans/test/utils/tools.py:557: in run_cli
    self._handle_cli_result(command_line, assert_error=assert_error, error=error)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <conans.test.utils.tools.TestClient object at 0x7fffec36a370>
command = 'build .', assert_error = False, error = 1

    def _handle_cli_result(self, command, assert_error, error):
        if (assert_error and not error) or (not assert_error and error):
            if assert_error:
                msg = " Command succeeded (failure expected): "
            else:
                msg = " Command failed (unexpectedly): "
            exc_message = "\n{header}\n{cmd}\n{output_header}\n{output}\n{output_footer}\n".format(
                header='{:-^80}'.format(msg),
                output_header='{:-^80}'.format(" Output: "),
                output_footer='-' * 80,
                cmd=command,
                output=self.out
            )
>           raise Exception(exc_message)
E           Exception: 
E           ------------------------ Command failed (unexpectedly): ------------------------
E           build .
E           ----------------------------------- Output: ------------------------------------
E           Using lockfile: '/tmp/guix-build-conan-1.40.2.drv-0/tmppw7a8172conans/path with spaces/conan.lock'
E           Using cached profile from lockfile
E           conanfile.py: Calling build()
The Meson build system                                                                              
E           Version: 0.55.1
E           Source dir: /tmp/guix-build-conan-1.40.2.drv-0/tmppw7a8172conans/path with spaces
E           Build dir: /tmp/guix-build-conan-1.40.2.drv-0/tmppw7a8172conans/path with spaces/build
E           Build type: native build
E           Using 'PKG_CONFIG_PATH' from environment with value: '/gnu/store/9860f1abqj8wjjnwl8a9v54pdcc3bhgf-xz-5.2.4/lib/pkgconfig:/gnu/store/9w9jvy3bgjg4qaqmrij01nbppiccqr7c-python-3.8.2/lib/pkgconfig'
E           Using 'PKG_CONFIG_PATH' from environment with value: '/gnu/store/9860f1abqj8wjjnwl8a9v54pdcc3bhgf-xz-5.2.4/lib/pkgconfig:/gnu/store/9w9jvy3bgjg4qaqmrij01nbppiccqr7c-python-3.8.2/lib/pkgconfig'
E           Project name: tutorial
E           Project version: undefined
E           C++ compiler for the host machine: c++ (gcc 7.5.0 "c++ (GCC) 7.5.0")
E           C++ linker for the host machine: c++ ld.bfd 2.34
E           Host machine cpu family: x86_64
E           Host machine cpu: x86_64
E           Build targets in project: 2
E           
E           Found ninja-1.10.0 at /gnu/store/9z915ibzqr70gr7a3rxkd0ikrxk3kmqn-ninja-1.10.0/bin/ninja
E           ninja: Entering directory `/tmp/guix-build-conan-1.40.2.drv-0/tmppw7a8172conans/path with spaces/build'
E           [1/5] Compiling C++ object demo.p/main.cpp.o
E           [2/5] Compiling C++ object libhello.so.p/hello.cpp.o
E           FAILED: libhello.so.p/hello.cpp.o 
E           c++ -Ilibhello.so.p -I. -I.. -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -g -fPIC -MD -MQ libhello.so.p/hello.cpp.o -MF libhello.so.p/hello.cpp.o.d -o libhello.so.p/hello.cpp.o -c ../hello.cpp
E           ../hello.cpp: In function ‘int hello()’:
E           ../hello.cpp:94:46: error: ‘TEST_DEFINITION1’ was not declared in this scope
E                std::cout << "   TEST_DEFINITION1: " <<  TEST_DEFINITION1 << "\n";std::cout << "   TEST_DEFINITION2: " <<  TEST_DEFINITION2 << "\n";
E                                                         ^~~~~~~~~~~~~~~~
E           ../hello.cpp:94:112: error: ‘TEST_DEFINITION2’ was not declared in this scope
E                std::cout << "   TEST_DEFINITION1: " <<  TEST_DEFINITION1 << "\n";std::cout << "   TEST_DEFINITION2: " <<  TEST_DEFINITION2 << "\n";
E                                                                                                                           ^~~~~~~~~~~~~~~~
E           ninja: build stopped: subcommand failed.
E           Found runner: /gnu/store/9z915ibzqr70gr7a3rxkd0ikrxk3kmqn-ninja-1.10.0/bin/ninja
E           ERROR: conanfile.py: Error in build() method, line 24
E               meson.build()
E               ConanException: Error 1 while executing meson compile -C "/tmp/guix-build-conan-1.40.2.drv-0/tmppw7a8172conans/path with spaces/build"
E           
E           --------------------------------------------------------------------------------

conans/test/utils/tools.py:592: Exception
________________________ MesonToolchainTest.test_build _________________________
[gw22] linux -- Python 3.8.2 /gnu/store/f8s95qc6dfhl0r45m70hczw5zip0xjxq-python-wrapper-3.8.2/bin/python

self = <conans.test.functional.toolchains.meson.test_meson.MesonToolchainTest testMethod=test_build>

    def test_build(self):
        hello_h = gen_function_h(name="hello")
        hello_cpp = gen_function_cpp(name="hello", preprocessor=["STRING_DEFINITION"])
        app = gen_function_cpp(name="main", includes=["hello"], calls=["hello"])
    
        self.t.save({"conanfile.py": self._conanfile_py,
                     "meson.build": self._meson_build,
                     "meson_options.txt": self._meson_options_txt,
                     "hello.h": hello_h,
                     "hello.cpp": hello_cpp,
                     "main.cpp": app})
    
        self.t.run("install . %s" % self._settings_str)
    
        content = self.t.load("conan_meson_native.ini")
    
        self.assertIn("[project options]", content)
        self.assertIn("STRING_DEFINITION = 'Text'", content)
        self.assertIn("TRUE_DEFINITION = true", content)
        self.assertIn("FALSE_DEFINITION = false", content)
        self.assertIn("INT_DEFINITION = 42", content)
        self.assertIn("ARRAY_DEFINITION = ['Text1', 'Text2']", content)
    
        self.assertIn("[built-in options]", content)
        self.assertIn("buildtype = 'release'", content)
    
        self.t.run("build .")
        self.t.run_command(os.path.join("build", "demo"))
    
>       self.assertIn("hello: Release!", self.t.out)
E       AssertionError: 'hello: Release!' not found in main: Debug!
E         main __x86_64__ defined
E         main _GLIBCXX_USE_CXX11_ABI 1
E         main __cplusplus201402
E         main __GNUC__7
E         main __GNUC_MINOR__5
E       hello: Debug!
E         hello __x86_64__ defined
E         hello _GLIBCXX_USE_CXX11_ABI 1
E         hello __cplusplus201402
E         hello __GNUC__7
E         hello __GNUC_MINOR__5
E          STRING_DEFINITION:

conans/test/functional/toolchains/meson/test_meson.py:86: AssertionError
```
Any idea?

Thanks!

--- a/conans/test/functional/toolchains/meson/_base.py
+++ b/conans/test/functional/toolchains/meson/_base.py
@@ -9,6 +9,8 @@ from conans.test.utils.tools import TestClient
 
 @pytest.mark.tool_meson
 @pytest.mark.skipif(sys.version_info.major == 2, reason="Meson not supported in Py2")
+@pytest.mark.skipif(platform.system() not in ("Darwin", "Windows", "Linux"),
+                    reason="Not tested for not mainstream boring operating systems")
 class TestMesonBase(unittest.TestCase):
     def setUp(self):
         self.t = TestClient()

